src/workerd/io/io-context.c++

```diff
@@ -1155,6 +1159,17 @@ void IoContext::runImpl(Runnable& runnable,
       }
     }
   });
+
+  if (!isCurrentNull()) {
+    KJ_LOG(ERROR, "IoContext not-null before running PumpMessageLoop()");
+  } else {
+    worker->runInLockScope(lockType, [&](Worker::Lock& lock) {
+        jsg::Lock& js = lock;
+        auto& system = const_cast<jsg::V8System&>(js.getV8System());
+        KJ_DBG(js.v8Isolate);
+        while (v8::platform::PumpMessageLoop(&system.getDefaultPlatform(), js.v8Isolate, v8::platform::MessageLoopBehavior::kDoNotWait)) {}
```

Comment:
hmm... what is it you're looking to accomplish here? Manually pumping the message loop is quite unusual. Is the idea here to try to force the finalization registry callback to run more predictably?

Reply:
We don't pump the message loop anywhere yet, and the Finalizers will never by invoked unless we manually pump it.
Deno had the same issue sometime back: https://github.com/denoland/deno/issues/7674#issuecomment-868925712

Reply:
Our failure to pump the message loop is also what led to the memory leak with liftoff code flushing in a recent V8 update. Apparently we were supposed to be doing this all along and somehow got away without it.